pyspecProcScripts/phasing.py

```diff
@@ -237,32 +237,50 @@ def hermitian_function_test(
     orig_dt = s.get_ft_prop(direct, "dt")
     if not s.get_ft_prop(direct):
         s.ft(direct)
-    s.ift(direct, pad=1024 * 4)
+    s.ift(direct, pad=ndshape(s)[direct]*50)
     new_dt = s.get_ft_prop(direct, "dt")
     non_aliased_range = r_[aliasing_slop,-aliasing_slop]*int(orig_dt/new_dt)
     if aliasing_slop > 0:
         s = s[direct,non_aliased_range[0]:non_aliased_range[1]]
+    s_envelope = s.C.mean_all_but(direct).run(abs)
     if fl is not None:
         fl.push_marker()
         fig, ax_list = subplots(2, 2, figsize=(15, 15))
         fl.next("Hermitian Function Test Diagnostics", fig=fig)
-        fl.plot(abs(s), ax=ax_list[0, 0], human_units=False)
+        fl.plot(s_envelope, ax=ax_list[0, 0], human_units=False)
         ax_list[0, 0].set_title("Data with Padding")
-    signal_range = abs(s).mean_all_but(direct).contiguous(lambda x: x > 0.5 * abs(x).data.max())
-    start_sig = signal_range[0][0]
-    end_sig = signal_range[0][-1]
-    peak = s[direct:(signal_range[0][0],signal_range[0][-1])].mean_all_but(direct).argmax()
-    extension_right = end_sig-peak.data
-    extension_left = peak.data - start_sig
-    if extension_right < extension_left:
-        start_sig = peak.data - extension_right
-        s = s[direct:(start_sig,end_sig)]
-    else:    
-        s = s[direct:(start_sig,end_sig+(extension_right*3))]
+    #s_envelope.convolve(direct, orig_dt*3)
+    #if fl is not None:
+    #    fl.plot(s_envelope, ax=ax_list[0, 0], human_units=False)
+    #peak_triple = abs(s).mean_all_but(direct).convolve(direct,orig_dt*3).contiguous(lambda x: x > 0.25 * x.data.max())[0,:]
+    peak_triple = s_envelope.contiguous(lambda x: x > 0.25 * x.data.max())[0,:]
+    # {{{ the peak triple gives the left and right thresholds, with the
+    #     peak max in the middle
+    #     we move the left point as needed to make sure that peak is in
+    #     the first half of the resulting slice
+    peak_triple = r_[ peak_triple[0],
+            s[direct:peak_triple].mean_all_but(direct).run(abs).argmax().item(),
+            peak_triple[1] ]
     if fl is not None:
+        for j in range(3):
+            ax_list[0, 0].axvline(x=peak_triple[j], ls=':', color='k', alpha=0.5, linewidth=2)
```

Comment:
your code was choosing bad ranges and I needed to do this to diagnose what was going on.

Reply:
(they were frequently only a very small slice of the signal)

Reply:
(and the cost function relies on averaging the rms from many points)